cli/commands/scaffold/action.go

```diff
@@ -166,6 +186,19 @@ func Run(ctx context.Context, opts *options.TerragruntOptions, moduleURL, templa
 
 	vars["sourceUrl"] = moduleURL
 
+	// Only set these if the `vars` map doesn't already have them set
+	if _, found := vars["EnableRootInclude"]; !found {
+		vars["EnableRootInclude"] = !opts.ScaffoldNoIncludeRoot
+	} else {
+		opts.Logger.Warnf("The EnableRootInclude variable is already set in the var flag(s). The --%s flag will be ignored.", NoIncludeRoot)
+	}
+
+	if _, found := vars["RootFileName"]; !found {
+		vars["RootFileName"] = opts.ScaffoldRootFileName
+	} else {
+		opts.Logger.Warnf("The RootFileName variable is already set in the var flag(s). The --%s flag will be ignored.", RootFileName)
+	}
```

Comment:
```go
const rootFileNameVarName = "RootFileName"
```

```suggestion
	if _, found := vars[rootFileNameVarName]; !found {
		vars[rootFileNameVarName] = opts.ScaffoldRootFileName
	} else {
		opts.Logger.Warnf("The %s variable is already set in the var flag(s). The --%s flag will be ignored.", rootFileNameVarName, RootFileName)
	}
```

Consider using a constant. This way we will be sure that the variable name in the message log will always be up-to-date.